### ISSUE TITLE
Include the hypcap features

### ISSUE THREAD

**Issue by dbitouze** [latex3/hyperref#154]
The package `hypcap` "tries a solution of the problem with `hyperref`, that links to floats points below the caption and not at the beginning of the float." It would be nice for `hyperref` to include the `hypcap` features.

**Comment by u-fischer** [comment]
Well this is not something that can done automatically in all cases: With a normal float with one caption, the default behaviour of hypcap is fine, but if more captions are involved it requires that the user understand what happens, and adds more \capstart commands in the right places. 

Beside this: the long term plan is to remove patches from hyperref, not to add more ;-).  So this will have to wait until there is sensible hook  inside the floats, that can be used to add the destination.

**Comment by dbitouze** [comment]
>Well this is not something that can done automatically in all cases: With a normal float with one caption, the default behaviour of hypcap is fine, but if more captions are involved it requires that the user understand what happens, and adds more `\capstart` commands in the right places.

OK but I guess 90 % of floats just have only one caption :)

> Beside this: the long term plan is to remove patches from `hyperref`, not to add more ;-).

This is not a patch request, just a feature request ;)

> So this will have to wait until there is sensible hook inside the floats, that can be used to add the destination.

So a very long term plan, I guess :$

**Comment by u-fischer** [comment]
> OK but I guess 90 % of floats just have only one caption :)

Yes, but hypcap has to redefine caption to avoid that it sets another destination, and this could break the other 10% floats.  



> So a very long term plan, I guess :$

Not really. Now that the hook management exists, more hooks can and will be added.
